src/chainstate/burn/db/sortdb.rs

```diff
@@ -950,17 +968,24 @@ impl <'a> SortitionHandleConn <'a> {
         SortitionDB::get_block_commit_parent(self, block_height, vtxindex, &self.context.chain_tip)
     }
 
+    /// Get a block commit by txid. In the event of a burnchain fork, this may not be unique.
+    ///   this function simply returns one of those block commits: only use data that is
+    ///   immutable across burnchain/pox forks, e.g., parent block ptr,  
+    pub fn get_block_commit_by_txid(&self, txid: &Txid) -> Result<Option<LeaderBlockCommitOp>, db_error> {
```

Comment:
Should we start indexing block-commit txids in the MARF in order to make sure that only valid block-commits can ever be loaded?  Either that, should we use a representation for block-commit data that is guaranteed to be fork-agnostic?

Reply:
I don't think so. None of the usage of these commit fetches depends on these lookups for fork-consistency, and avoiding the overhead of additional MARF indexing is probably for the best.